### ISSUE TITLE
The Chinese (Cantonese) voice is broken

### ISSUE THREAD

**Issue by kara-louise** [nvaccess/nvda#10418]
As the title says, this voice is broken.
After a quick look at eSpeak NG's source files, there are a few that start with zhy. I believe these, when compiled, create the Cantonese voice. I couldn't find any zhy files in C:\Program Files (x86)\NVDA\synthDrivers\espeak-ng-data.
### Steps to reproduce:
1. Open NVDA's speech settings page.
2. In the voice combo box, select "Chinese (Cantonese")
### Actual behavior:
NVDA doesn't speak.
### Expected behavior:
NVDA should speak with the selected voice.
#### NVDA installed/portable/running from source:
Installed, but this also happens with portable installations. 
#### NVDA version:
2019.2.1

#### Windows version:
10.0.18362 Build 18362
#### Does the issue still occur after restarting your PC?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
latest snapshot build (alpha-19004,ab429417) at time of writing.
Same issue occurs.

**Comment by Brian1Gaff** [comment]
Would you be able to see if the sapi 5 version of espeak has the files? 
 Just wondered if it might be systemic.
 Brian

bglists@blueyonder.co.uk 
Sent via blueyonder. 
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- O

**Comment by kara-louise** [comment]
eSpeak NG isn't currently SAPI 5 compatible. But I installed the latest stable windows build ( 1.49.2 3ed34d3 released on Sep 24, 2017), and it does include a "zhy_dict" file.
If placed in synthDrivers\espeak-ng-data,, I get speech when selecting the Cantonese voice, but Chinese characters are read as "Chinese letter x" Where x is a number.
I lack the skills to build eSpeak NG from source so can't check the master branch as it stands currently.
So the build included with NVDA's missing a file.

**Comment by leonardder** [comment]
I can not reproduce this issue.

@feerrenrut May be we should consider a bump of espeak?

**Comment by zstanecic** [comment]
Maybe we may consider updating espeak to latest master?

There were some updates out there for particular languages.

**Comment by feerrenrut** [comment]
`zhy` dictionary compilation is intentionally excluded.  
If someone wishes to investigate further, see `nvdaHelper/espeak/sconscript` 
Under the comment `#Compile all dictionaries`
Change the line:
`missingDicts=['zhy', ] #'mt','tn','tt']` 
to:
`missingDicts = []`
And try to build, the errors are:
```
espeak_compileDict_buildAction(["include\espeak\espeak-ng-data\zhy_dict"], ["include\espeak\dictsource\zhy_rules"])
espeak_compileDict_action: failed to switch to language b'zhy'
scons: *** [include\espeak\espeak-ng-data\zhy_dict] Error 1
```

**Comment by Qchristensen** [comment]
I've had correspondence from another user reporting this.  Did we work out what the issue is with the language?

**Comment by kara-louise** [comment]
The voice's still broken in alpha-20758,0098a31a.

**Comment by feerrenrut** [comment]
I have a potential fix for this, could you please test [this build](https://ci.appveyor.com/api/buildjobs/wv6l2lddhqsmt13c/artifacts/output%2Fnvda_snapshot_pr12370-22601%2C6da721c8.exe) from #12370

**Comment by kara-louise** [comment]
@feerrenrut That seems to work.

**Comment by feerrenrut** [comment]
Fixed via  fca3c2d in PR Fix compilation of espeak dictionary for language zhy #12370
